spark3/src/main/java/org/apache/iceberg/spark/SparkFilters.java

```diff
@@ -198,6 +200,10 @@ private static Object convertLiteral(Object value) {
       return DateTimeUtils.fromJavaTimestamp((Timestamp) value);
     } else if (value instanceof Date) {
       return DateTimeUtils.fromJavaDate((Date) value);
+    } else if (value instanceof Instant) {
```

Comment:
Do we know passed in `Instant` always correspond to type `timestamptz` but not `date`?

Reply:
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala#L81

> LGTM, though I'm not a spark expert; some minor questions. Do we know if this only impact spark3 but not 2?

Yep the patch to add support for these literals only exists in Spark3
https://issues.apache.org/jira/browse/SPARK-26902